lib/modules/home/home_view.dart

```diff
@@ -1,11 +1,37 @@
 part of 'home_module.dart';
 
-abstract class HomeViewDelegate {}
+abstract class HomeViewDelegate implements AlertViewMixinDelegate {
+  BehaviorSubject<void> get stateDidInit;
+
+  BehaviorSubject<SurveyInfo> get showDetailButtonDidTap;
+
+  BehaviorSubject<void> get didSwipeDown;
```

Comment:
@markgravity I have revised your implementation in `ViewDelegate`. Currently you are using `BehaviorSubject` to call the action in Presenter from View. For example:
- Call in View: `delegate?.stateDidInit.add(null);`
- Listen function in Presenter:
```dart
stateDidInit.voidListen(_stateDidInit).addTo(disposeBag);
...
void _stateDidInit() {
    view.beginAnimation();
    interactor.validateAuthentication();
  }
```

I think creating `BehaviorSubject` and call directly in View is not the best approach in term of clean code. Instead, we can create a function like `stateDidInit()` in `ViewDelegate`, then override it in Presenter and perform the logic. Ex:
- Create function in ViewDelegate:
![image](https://user-images.githubusercontent.com/60863885/120427225-bb9e6800-c39b-11eb-9699-89ed3b598265.png)
- Need to call? Here it is:
![image](https://user-images.githubusercontent.com/60863885/120427352-f43e4180-c39b-11eb-99e4-f815373db52a.png)
- Implement in Presenter:
![image](https://user-images.githubusercontent.com/60863885/120427421-159f2d80-c39c-11eb-819e-c5cf7c45944c.png)